front/app/utils/moduleUtils.tsx

```diff
@@ -450,11 +457,12 @@ export const insertConfiguration =
     insertBeforeName,
   }: InsertConfigurationOptions<T>) =>
   (items: T[]): T[] => {
-    const itemAlreadyInserted = items.some(
+    const itemsClone = cloneDeep(items);
```

Comment:
Just curious why we needed the deep clone in this situation? Is there a risk this takes a long time if there are many items (e.g. where insertConfiguration is used in post manager)?

Reply:
The short answer is to avoid side effects. Those always make bugs harder to find. In this case, we don't alter what the function takes in but return a new data structure to be used